api/workload/v2/workload.go

```diff
@@ -0,0 +1,244 @@
+package workload
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spiffe/spire/proto/spire/api/workload"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	// DefaultAgentAddress is the default GRPC address to contact the spire agent at.
+	DefaultAgentAddress = "unix:///tmp/agent.sock"
+
+	// EnvVarAgentAddress is the environment variable name where the Workload API address may be configured.
+	EnvVarAgentAddress = "SPIFFE_ENDPOINT_SOCKET"
+
+	_unixPathPrefix = "unix://"
+)
+
+// X509SVIDs is an X.509 SVID response from the SPIFFE Workload API.
+type X509SVIDs struct {
+	// SVIDs is a list of X509SVID messages, each of which includes a single
+	// SPIFFE Verifiable Identity Document, along with its private key and bundle.
+	SVIDs []*X509SVID
+
+	// CRL is a list of revoked certificates.
+	// Unimplemented.
+	CRL *pkix.CertificateList
+
+	// FederatedBundles are CA certificate bundles belonging to foreign Trust Domains
+	// that the workload should trust, keyed by the SPIFFE ID of the foreign domain.
+	// Unimplemented.
+	FederatedBundles map[string]*x509.CertPool
```

Comment:
hmm, should this also be []*x509.Certificate instead?

Reply:
hmm yes i think so